lib/baseUrl.ts

```diff
@@ -0,0 +1 @@
+export default process.env.NEXT_PUBLIC_BASE_URL || 'http://localhost:3000';
```

Comment:
This change requires a new env variable to be configured. If you want, we can change it to `NEXT_PUBLIC_VERCEL_URL` which is automatically exposed by Vercel and would remove the need for a new env variable.

I've used a different name to be agnostic about the provider since a Next.js app can be deployed to many places.

Reply:
We can go with `NEXT_PUBLIC_BASE_URL` as it simplifies the process while self-hosting. 

If we were to use `NEXT_PUBLIC_VERCEL_URL`, changes have to be made in the code too while self hosting.